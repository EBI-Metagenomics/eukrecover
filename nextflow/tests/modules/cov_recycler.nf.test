nextflow_process {

    name "Test module coverage recycler"
    script "modules/cov_recycler.nf"
    process "COVERAGE_RECYCLER"

    test("coverage_recycler") {
        when {
            process {
                """
                input[0] = channel.from([
                    tuple(
                    "ERZ857996",
                    [file("${baseDir}/tests/resources/binning/maxbin2_bins/bin.0.fa"),
                     file("${baseDir}/tests/resources/binning/maxbin2_bins/bin.1.fa")],
                    file("${baseDir}/tests/resources/metabat_depth.txt"))
                    ])
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}