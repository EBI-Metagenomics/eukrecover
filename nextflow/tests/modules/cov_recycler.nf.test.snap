{
    "coverage_recycler": {
        "content": [
            {
                "coverage_dir": [
                    [
                        "ERZ857996",
                        [
                            [
                                "bin.0.fa_MAGcoverage.txt:md5,9582532556e884d46a16b68648008f94",
                                "coverage.tab:md5,d4036f88b34d526c7d697450c27b7d62"
                            ],
                            [
                                "bin.1.fa_MAGcoverage.txt:md5,141f8dd1795a7cf0e36e90557e11dd54",
                                "coverage.tab:md5,f93b77d01ac66ba966cb7e70f316ce02"
                            ],
                            "contigs2bins.txt:md5,6fde0ca6d3ea8c9d67a803246605a362"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-06-15T11:52:53+0000"
    }
}